gadopt/momentum_equation.py

```diff
@@ -115,6 +116,11 @@ def residual(self, test, trial, trial_lagged, fields, bcs):
                 # here we need only the third term, because we assume jump_u=0 (u_ext=u)
                 # the provided stress = n.(mu.stress_tensor)
                 F += dot(-phi, bc['stress']) * self.ds(id)
+            if 'free_surface_stress' in bc:
+                # add the free surface stress
+                free_surface_stress = bc['free_surface_stress']
+                F -= dot(-phi, free_surface_stress * n) * self.ds(id)
```

Comment:
From the implementation it looks like `'free_surface_stress'` is just a "normal_stress" should we call it that?

Similar to the current checks at the end of this subroutine, you probably want to checks against the combination of `u` or `un` with "normal_stress"/"free_surface_stress". Setting a tangential u component with a free surface _might_ make sense (I vaguely recall having tried that in Fluidity for some case) but we can add it if/when someone needs it.

Reply:
I've renamed "free_surface_stress" to "normal_stress" (I think this is clearer too...!)

Reply:
and I've also added some checks at the end for combinations of u/un and normal  stress